docs/messaging/messaging-metrics.md

```diff
@@ -0,0 +1,167 @@
+
+**[8]:** This should be the IP/hostname of the broker (or other network-level peer) this specific message is sent to/received from.
+
+**[9]:** When observed from the client side, and when communicating through an intermediary, `server.port` SHOULD represent the server port behind any intermediaries, for example proxies, if it's available.
+
+`error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. |
+<!-- endsemconv -->
+
+## Producer metrics
+
+### Metric: `messaging.publish.duration`
+
+This metric is [required][MetricRequired].
+
+When this metric is reported alongside a messaging publish span, the metric value SHOULD be the same as the corresponding span duration.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advice)
+of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10 ]`.
+
+<!-- semconv metric.messaging.publish.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.publish.duration` | Histogram | `s` | Measures the duration of publish operation. |
+<!-- endsemconv -->
+
+### Metric: `messaging.publish.messages`
+
+This metric is [required][MetricRequired] when the messaging system supports batch publishing. It's [opt-in][MetricOptIn] when the messaging system does not support batch publishing, since the message count can be derived from the `messaging.publish.duration` histogram.
+
+_Note: The need to report `messaging.publish.messages` depends on the messaging system capabilities and not application scenarios or client library limitations. For example, RabbitMQ does not support batch publishing and corresponding instrumentations don't need to report `messaging.publish.messages`. Kafka supports both, single and batch publishing, and instrumentations MUST report `messaging.publish.messages` counter regardless of application scenarios or APIs available in the client library._
+
+<!-- semconv metric.messaging.publish.messages(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.publish.messages` | Counter | `{message}` | Measures the number of published messages. |
+<!-- endsemconv -->
+
+## Consumer metrics
+
+### Metric: `messaging.receive.duration`
+
+This metric is [required][MetricRequired] for poll-based receive operations.
```

Comment:
```suggestion
This metric is [required][MetricRequired] for pull-based receive operations.
```
Or maybe a bit more verbose? 

```suggestion
This metric is [required][MetricRequired] for operations that are initiated by the application code (pull-based).
```

Reply:
any reason to use `pull`? Polling and poll is frequently used in the context of queues.

Reply:
oh, I see, we already use push in other places.